src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -284,6 +284,17 @@
       <LatestCommit>N/A</LatestCommit>
     </PropertyGroup>
 
+
+    <!-- If things match expected patterns, create $(GitBranchName) from $(RawGitBranchName) -->
+    <Message Text="GIT Branch Name as seen by git describe: '$(RawGitBranchName)'" /> 
+    <PropertyGroup Condition="$(RawGitBranchName.StartsWith('he:ads/'))">
+      <GitBranchName>$(RawGitBranchName.SubString(6))</GitBranchName>
+    </PropertyGroup>
+    <PropertyGroup Condition="$(RawGitBranchName.StartsWith('remotes/origin/'))">
```

Comment:
Do you think it would be worth it to make this generic rather than assuming the remote's called `origin`?

Reply:
MSBuild does have search and substitute, so it is not that big of a deal to do this, but I do wonder whether it is interesting.   Build servers are likely to be the simplest case (clone some other repo in the simplest way possible), and thus you will get 'origin'   Given that if we are wrong it is not fatal (no worse then what we have now) I am OK with reacting as needed.  If we know it is going to be an issue, I can do it now...  Do you think we will have a problem?

Reply:
That's true, I agree this almost definitely won't be a problem for official builds. It can be fixed for dev builds if somebody cares, but I think that's unlikely.